ESPLoader.js

```diff
@@ -1586,6 +1587,29 @@ class ESPLoader {
         this.log("Device: "+((flashid >> 8) & 0xff).toString(16) + flid_lowbyte.toString(16));
         this.log("Detected flash size: " + this.DETECTED_FLASH_SIZES[flid_lowbyte]);
     }
+
+    hard_reset = async() => {
+        this.transport.setRTS(true);  // EN->LOW
+        await this._sleep(100);
+        this.transport.setRTS(false);
+    }
+
+    soft_reset = async() => {
```

Comment:
When is a soft reset used versus a hard reset?

Reply:
A soft reset is used when the circuit doesn't utilise the RTS/DTR lines. As in an ExpressLRS receiver.